src/supabase/yourFollowingFunctions.js

```diff
@@ -0,0 +1,116 @@
+import { supabase } from "./supabaseClient";
+import { getFollowing } from "../api/twitter";
+import { 
+    handleProfileOnLogin, 
+    doesProfileExist, 
+    doesNewUserExist,
+    deductNewUser,
+    checkNewUsers
+} from "./profileFunctions";
+
+/*
+    Main function to update following of users within database.
+    By default, it prioritizes brand new users. This helps to push
+    their data into the system first.
+    Those in system needs to wait for new users to clear.
+*/
+export const checkLimit = async (session) => {
+    // check the amount of awaiting new users in system
+    const newUsers = await checkNewUsers();
+    const doesExistNewUser = await doesNewUserExist(session);
+    const doesExistProfile = await doesProfileExist(session);
+    if (doesExistProfile && !doesExistNewUser) { // user already exists
+        const profileTime = await profileTimeDifference(session)
+        //console.log(profileTime)
+        if (newUsers == 0 && profileTime >= 30.0) {
+            const updateDatabase = await updateFollowings(session);// tries to update user's following
+            const updatingTime = await updateProfileTime(session);// updates the update time in table
+            return true;
+        }
+    } else { // its a new user
+        const addingUser = await handleProfileOnLogin(session); // add user if he is not already in
+        const updateDatabase = await updateFollowings(session); // tries to update user's following
+        const deletingNewUser = await deductNewUser(session); // deletes new users
```

Comment:
Currently the update function is called within `groupSelectionImprovedV2.js`. The try catch error block is meant to handle an error if it fails to update the profile. On error, if the user exists within new user table, it sets useState for `following` to false. Then in yourFollowing, if the `following` is false, it will continue to show the loading button.